ingest/src/main/java/com/microsoft/azure/kusto/ingest/IngestionProperties.java

```diff
@@ -234,7 +234,7 @@ public void setDataFormat(String dataFormatName) {
         }
     }
 
-    public String getDataFormat() {
+    public String getDataFormatStr() {
```

Comment:
For sure, but this is a unique situation, where otherwise we're "lying" about the return type. In this case, since we have a data type called `DataFormat`, one would expect `getDataFormat()` to return an object of that type. My approach is that user of even a class API should never be surprised, which is why I added the "Str" to the end.

Reply:
Ok, but we should note this is a breaking change (renaming a public method).
If we're ok with this, I think we should not store DataFormat as an str, but as a DataFormat like we do with the tags.

Reply:
Yeah I'm trying to get this in davka before the release I want to do this week to 3.0.0.

I'm not sure what you're referring to regarding tags. This field is part of `AdditionalProperties`, which is a Map<String, String>. `Tags` is a `List<String>`. I think it's intentional keeping it a String because we don't need the entire object to be serialized and sent over the wire.

Reply:
I don't understand what do you mean.

Other properties, such as tags and ingestionmapping, have their interface as their actual type (list<string> and IngestionMapping), and when they're serialized we turn them into the proper strings.

DataFormat should follow the same logic - the interface will be with the type DataFormat, and when we serialize it we'll turn it into string.

Reply:
The `IngestionMapping` object is all relevant because it requires sending all that data on the wire. There's nothing in that object the service doesn't need.

But everywhere else, we minimize what we send to only what's needed. For example, I see `private List<String> dropByTags`, `private List<String> ingestByTags` and `private List<String> additionalTags`. I don't see any unnecessary object info being sent over the wire.

The `DataFormat` is stored as a record in `private Map<String, String> additionalProperties`. That's just a String to String mapping. We **could** move the DataFormat to be in its own field, but I'm not seeing any additional info from its enum object we'd **want** to include.

Reply:
Yes, I meant moving DataFormat into its own field rather than additionalProperties. 
The advantage we get is type safety.
We don't have to parse it whenever we use it, and it is less error prone.

Reply:
That would be much more than a breaking change - we'd have to change it in every SDK, and on the service side!

Reply:
No we won't? 
Instead of having it in additionalFields, just have a DataFormat typed field, and do 
fullAdditionalProperties.put("format", d.name());
Why would we need to change anything else?

Reply:
So you're suggesting maintaining in `IngestionProperties` both the full enum (I still don't understand what from the enum would be useful to send other than the name) and the name, and sending both over the wire?